scripts/update_python.py

```diff
@@ -32,29 +32,35 @@
 from subprocess import check_call
 from zip import unzip_cmd, zip_cmd
 
-version = '3.9.2'
+version = '3.13.0'
```

Comment:
Do we need to upgrade to such a recent version?  i.e. what as the first version of which arm64 packages are available.

The slight downside of using a bleeding edge version of python like this is that we accidentally start depending of very new python features that folks on older/stable versions might not have.

Reply:
I was pondering about this too. My thinking here is that since we do these updates so rarely, maybe we would for once update to a modern version.

Python 3.9, 3.10 and 3.11 will no longer receive bugfixes to any issues one might run into. I don't think it makes sense to update to any of these versions, even if one of these would be the earliest ones to support Windows on ARM64.

Python 3.14 is in feature development. So we definitely will not want to update to that branch.

Python 3.12 and Python 3.13 have finished feature development and are actively receiving bufixes. It feels that either of these would be the ones to focus on.

I.e.

![image](https://github.com/user-attachments/assets/62dc5625-3710-4950-a019-328770e92815)

I picked 3.13 since it is newer, and I could not find any mentions of OS support cutoff differences between 3.12 and 3.13.

> we accidentally start depending of very new python features that folks on older/stable versions might not have.

3.13 is stable. (or rather, actually, every python version is broken in different ways. But Emscripten test suite is fortunately evolved to be good at finding ways that Python works or doesn't work out)

We do bundle python for the very purpose to normalize which version we provide to people? So this covers Windows and macOS users. On Linuxes where this currently is not the case, I understand updating to a newer version should be possible easier with their package managers?

Reply:
I suppose the other alternative could be [3.12.7](https://www.python.org/downloads/release/python-3127/) . At the end of the day I don't feel that strongly, but generally would strive to standardize to latest version, especially if Emscripten test suite will pass with 3.13.0. (Testing that out on Windows, although I am seeing Windows failures already before this update, making it a bit more time consuming to compare)

Reply:
sgtm

Reply:
;-)  https://www.python.org/downloads/release/python-3131/
```suggestion
version = '3.13.1'
```